### PR TITLE
fixes url encoding issue for a special character

### LINKED ISSUE #10351: Github link formatting issue
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I am running `fastlane deliver download_metadata` to download the meta data from iTunes Connect. I getting error following error in the end.

<details>

```
Looking for related GitHub issues on fastlane/fastlane...
Search query: undefined method `version' for nil

URL: https://api.github.com/search/issues?q=undefined%20method%20%60version'%20for%20nil+repo:fastlane/fastlane
➡️  undefined method `details' for nil:NilClass
    https://github.com/fastlane/fastlane/issues/9947 [open] 5 💬
    a week ago

➡️  fastlane run add_git_tag: undefined method `delete' for nil:NilClass
    https://github.com/fastlane/fastlane/issues/10075 [open] 0 💬
    4 weeks ago

➡️  Actions::EnsureGitStatusCleanAction.run(nil) Crashes on Dirty Branch
    https://github.com/fastlane/fastlane/issues/9747 [open] 4 💬
    a week ago

and 239 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60version'%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/fastlane-2.56.0/deliver/lib/deliver/commands_generator.rb:165:in `block (2 levels) in run': [!] undefined method `version' for nil:NilClass (NoMethodError)
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/fastlane-2.56.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:64:in `run!'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/fastlane-2.56.0/deliver/lib/deliver/commands_generator.rb:176:in `run'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/fastlane-2.56.0/deliver/lib/deliver/commands_generator.rb:12:in `start'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/fastlane-2.56.0/fastlane/lib/fastlane/cli_tools_distributor.rb:58:in `take_off'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0@global/gems/fastlane-2.56.0/bin/fastlane:20:in `<top (required)>'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0/bin/fastlane:22:in `load'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0/bin/fastlane:22:in `<main>'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `eval'
	from /Users/rtayal/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `<main>'
```
</details>

The issue I am reporting here is that the Github links contain `'` character which translates wrong url to the ⌘ + click experience. When I click on the url `https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60version'%20for%20nil&type=Issues&utf8=✓`, it only takes the part before `'` which is `https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60version`. The rest of the part of the url is omitted.

Will create a pull request to fix this so clicking on the issue link will open the correct url.

### PR DESCRIPTION
**Pull request by RishabhTayal**
fixes: #10351

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/ui/github_issue_inspector_reporter.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@ def inspector_successfully_recieved_report(report, inspector)
       report.issues[0..(NUMBER_OF_ISSUES_INLINE - 1)].each { |issue| print_issue_full(issue) }
 
       if report.issues.count > NUMBER_OF_ISSUES_INLINE
+        report.url.sub! '\'', '%27'
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/ui/github_issue_inspector_reporter.rb]: Please surround method calls with `(` and `)`

**Review by KrauseFx — COMMENTED** [review]
Looks great, thanks, one little comment before it's ready

File: fastlane_core/lib/fastlane_core/ui/github_issue_inspector_reporter.rb (hunk 1 above)
Comment (RishabhTayal) [reply fastlane_core/lib/fastlane_core/ui/github_issue_inspector_reporter.rb]: done and pushed.

**Review by KrauseFx — APPROVED** [review]
Thanks 👍

**Pull request merged**